### ISSUE TITLE
Design: Update table search bar so the form field label is visible

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2535]
The search bar for the domains and domain requests table does not have a visual label when user input is in the form field. This is a failure section 508 trusted tester ID 3.3.2-label-provided.

![Image](https://github.com/user-attachments/assets/8fb29a9f-90a9-4727-b47e-fff4a9861a95)
![Image](https://github.com/user-attachments/assets/2b9a25b4-e7e0-4114-8189-2c711783866c)

### Acceptance criteria

- [x] Redesign the search bar component with a label that is visible at all times.
- [x] Review designs with design team, product, and dev team.
- [ ] Create a development ticket.

**Comment by AnnaGingle** [comment]
Reviewing designs at the design product/design sync. 
Designs

- [Option 1](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9478-45648&t=5F8SXMsr81fX2q0s-4): No design changes outside of adding the label
- [Option 2](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9478-6035&t=5F8SXMsr81fX2q0s-4), [Option 3](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9478-44646&t=5F8SXMsr81fX2q0s-4): Different ways of displaying the search bar, header, and export button.
